mrmustard/lab_dev/wires.py

```diff
@@ -176,6 +187,42 @@ def __init__(
         # Adds elements to the cache when calling ``__getitem__``
         self._mode_cache = {}
 
+    @cached_property
+    def adjoint(self) -> Wires:
+        r"""
+        New ``Wires`` object obtained by swapping ket and bra wires.
+        """
+        return Wires(
+            self.args[2], self.args[3], self.args[0], self.args[1], self.args[4], self.args[5]
+        )
+
+    @cached_property
+    def bra(self) -> Wires:
+        r"""
+        New ``Wires`` object with only bra wires.
+        """
+        ret = Wires(self.args[0], self.args[1], set(), set())
```

Comment:
Do you need two more empty sets here so the total number of sets is 6?

Reply:
Not really, tbh the last two empty sets aren't even needed. The reason is that it will default to `None` which is already treated as `set()` in the `__init__`.